### ISSUE TITLE
I had to restart the client to refresh the handlers for a check

### ISSUE THREAD

**Issue by FlorinAndrei** [sensu/sensu#806]
```
$ cat /etc/sensu/conf.d/check_foobar.json 
{
	"checks": {
		"foo_cpu": {
			"handlers": ["default", "pagerduty"],
			"pager_team": "pd_foobar_metrics",
			"command": "check-stats.rb -h localhost:8080 -t 'foobar.*.*.*.*.os.cpu.user' -w 30 -c 50 -p -5mins",
			"interval": 60,
			"subscribers": [ "foometrics" ]
		}
	}
}
```

sensu-0.14.0 on Ubuntu 14.04, installed from omnibus apt package.

Server, client, api, redis, rabbitmq, and graphite, are all on the same machine (I also have a bunch of sensu clients in other places connecting back here). All sensu config (client, server, api) is in /etc/sensu/conf.d.

check-stats.rb referenced in the JSON above checks metrics in graphite (-t is the graphite path).

For that check I changed the second handler from "pagerduty" to "slack", and I restarted sensu-server. Then I triggered a high CPU condition on another machine and waited. In due time, the check triggered, but the alert was sent to pagerduty. I tried a few more times, but got same result.

Only after I restarted sensu-client on this machine did the alert go to slack.

Isn't this definition server-related? Shouldn't its behavior change when I restart the server? Why did it change only when restarting the client?

**Comment by WolfgangBoremski** [comment]
I have had the same issue. I added a second handler to my check and had to restart the client to let him know about the new check-configuration. Depending on the number of clients this can be very anoying. The Server should push a new checks.json to the client if anything changes. 
This feels like a Bug

**Comment by portertech** [comment]
This is not a bug, it was intended to provide client specific overrides (snowflake) and enforcement.

**Comment by FlorinAndrei** [comment]
Sean, it is a bit confusing, though. The overrides are a useful feature, but I wish there was a way to more clearly separate server-side and client-side config bits.

**Comment by calebhailey** [comment]
Closing due to inactivity. Please feel free to create a new issue if this matter is not resolved. #monitoringlove
